zebra-chain/src/transaction/transparent.rs

```diff
@@ -23,18 +23,26 @@ pub struct OutPoint {
 
 /// A transparent input to a transaction.
 #[derive(Clone, Debug, Eq, PartialEq)]
-#[cfg_attr(test, derive(Arbitrary))]
-pub struct TransparentInput {
-    /// The previous output transaction reference.
-    pub previous_output: OutPoint,
-
-    /// Computational Script for confirming transaction authorization.
-    pub signature_script: Script,
-
-    /// Transaction version as defined by the sender. Intended for
-    /// "replacement" of transactions when information is updated
-    /// before inclusion into a block.
-    pub sequence: u32,
+pub enum TransparentInput {
+    /// A reference to an output of a previous transaction.
+    PrevOut {
+        /// The previous output transaction reference.
+        outpoint: OutPoint,
+        /// The script that authorizes spending `outpoint`.
+        script: Script,
+        /// The sequence number for the output.
+        sequence: u32,
+    },
+    /// New coins created by the block reward.
+    Coinbase {
+        /// The height of this block.
+        height: BlockHeight,
+        /// Approximately 100 bytes of data (95 to be safe).
+        /// XXX refine this type.
```

Comment:
@dconnolly's previous comment: https://github.com/ZcashFoundation/zebra/pull/235#discussion_r376612223

Making a `[u8; 100]` isn't quite right, because we need to have something that could be smaller, and in fact we don't know even how much smaller, because if we keep a single invariant on allowable `BlockHeight`s, we may need to restrict it to 95.  I'm not sure if there's a great way to encode this in the type system.